packages/projen/component/dir-env/src/dirEnv.ts

```diff
@@ -0,0 +1,278 @@
+	readonly contents?: string[]
+}
+
+export class DirEnv extends Component {
+	public static of(project: Project): DirEnv | undefined {
+		const isDirenv = (c: Component): c is DirEnv => c instanceof DirEnv
+		return project.components.find(isDirenv)
+	}
+
+	readonly #fileContents: string[] = []
+	public readonly file: TextFile
+
+	constructor(project: Project, public readonly options: DirEnvOptions = {}) {
+		super(project)
+		const { fileName = '.envrc' } = options
+		this.#validateFileName(fileName)
+		if (options.contents) {
+			this.#fileContents = options.contents
+		} else {
+			this.build().buildDefaultEnvRc()
+		}
+		this.file = new TextFile(project, fileName, {
+			readonly: true,
+			executable: true,
+		})
+	}
+
+	#validateFileName(fileName: string) {
+		const regex = new RegExp(/^\.envrc.*$/)
+		if (!regex.test(fileName)) {
+			throw new Error(`Invalid file name. Does not match ${regex.toString()}`)
+		}
+	}
+
+	/**
+	 * Append line to fileContents.
+	 * @param line
+	 * @param indent
+	 */
+	addLine(line: string, indent = 0) {
+		this.#fileContents.push(`${' '.repeat(indent)}${line}`)
+		return this
+	}
+
+	/**
+	 * Append lines to fileContents.
+	 * @param lines
+	 * @param indent
+	 */
+	addLines(lines: string[], indent = 0) {
+		lines.forEach((line) => this.addLine(line, indent))
+		return this
+	}
+
+	/**
+	 * Add an environment variable.
+	 * @param key
+	 * @param value
+	 */
+	addEnvVar(key: string, value: string) {
+		this.addLine(`export ${key}=${value}`)
+		return this
+	}
+
+	/**
+	 * Add an environment variable with a default value.
+	 * Useful for adding sample environment variables.
+	 * @param key
+	 * @param value
+	 */
+	addEnvVarWithDefault(key: string, value = '') {
+		this.addLine(`export ${key}="\${${key}:-${value}}"`)
+		return this
+	}
+
+	/**
+	 * Add a comment.
+	 */
+	addComment(comment: string) {
+		this.addLine(`# ${comment}`)
+		return this
+	}
+
+	/**
+	 * Add a command.
+	 */
+	addCommand(command: string, indent = 0) {
+		this.addLine(`${' '.repeat(indent)}${command}`)
+		return this
+	}
+
+	/**
+	 * Add a blank line.
+	 */
+	addBlankLine() {
+		this.addLine('')
+		return this
+	}
+
+	/**
+	 * Add a shebang.
+	 */
+	addSheBang() {
+		this.addLine('#!/usr/bin/env bash')
+		return this
+	}
+
+	/**
+	 * Add a direnv stdlib command.
+	 * @param command
+	 * @param args
+	 */
+	addStdLibCommand(command: DirEnvStdLibCommand, ...args: string[]) {
+		this.addLine(`${command} ${args.join(' ')}`)
+		return this
+	}
+
+	/**
+	 * Add status or error log using direnv log_status or log_error helpers.
+	 * @param type
+	 * @param message
+	 */
+	addLog(type: 'status' | 'error', message: string) {
+		const cmd =
+			type === 'status'
+				? DirEnvStdLibCommand.LOG_STATUS
+				: DirEnvStdLibCommand.LOG_ERROR
+		this.addStdLibCommand(cmd, message)
+		return this
+	}
+
+	/**
+	 * Add a use command.
+	 * @param use
+	 */
+	addUse(use: DirEnvUse) {
+		this.addStdLibCommand(DirEnvStdLibCommand.USE, use)
+		return this
+	}
```

Comment:
:thinking: I get the thought process of adding helpers for common commands...

But why just simplify `addStdLibCommand` to `addCommand`